### PR TITLE
ARROW-14127: [C++][Docs] Example of using compute function and output

### PR DESCRIPTION
**Pull request by bkmgit**

### CODE REVIEW

File: cpp/examples/arrow/compute_and_write_csv_example.cc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        auto schema = arrow::schema({arrow::field("a", arrow::int64()),
+                                     arrow::field("b", arrow::int64()),
+                                     arrow::field("c", arrow::boolean()),
+                                     arrow::field("d", arrow::boolean())  });
+        std::shared_ptr<arrow::Table> my_table = arrow::Table::Make(schema, {array_a,array_b,array_c,
+                                                           compared_array});
+
+        std::cout << "Table created" << std::endl;
+
+        // Write table to CSV file
```
Comment (cyb70289) [inline cpp/examples/arrow/compute_and_write_csv_example.cc:110]: Though it's an example, I think we should still be careful not to accidentally destroy a possible same named file under current directory.

Comment (bkmgit) [reply cpp/examples/arrow/compute_and_write_csv_example.cc:110]: Thanks for the feedback. The [dataset_documentation_example.cc ](https://github.com/apache/arrow/blob/master/cpp/examples/arrow/dataset_documentation_example.cc) creates a new directory for the files that are created. Can follow the same approach here.

Comment (bkmgit) [reply cpp/examples/arrow/compute_and_write_csv_example.cc:135]: @cyb70289 Have updated the example. Let me know if ok.

File: cpp/examples/arrow/compute_and_write_csv_example.cc [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+  std::cout << "Table created" << std::endl;
+
+  // Create a folder to output the data
+  std::string uri = argv[1];
+  std::string root_path;
+  auto fs = arrow::fs::FileSystemFromUri(uri, &root_path).ValueOrDie();
+  std::string base_path = root_path + "/csv_dataset";
+  std::cout << "Base path " << base_path << std::endl;
+  ARROW_RETURN_NOT_OK(fs->CreateDir(base_path));
```
Comment (cyb70289) [inline cpp/examples/arrow/compute_and_write_csv_example.cc:118]: appveyor ci error needs be fixed:
https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/40927445/job/rmy2wlpipu4xg6ny#L1201

Comment (bkmgit) [reply cpp/examples/arrow/compute_and_write_csv_example.cc]: @cyb70289 Appveyor build passes. Tests in C++ / AMD64 MacOS 10.15 C++  pass, but then it seems to time out.

Comment (bkmgit) [reply cpp/examples/arrow/compute_and_write_csv_example.cc]: @cyb70289 All tests pass.

Comment (cyb70289) [reply cpp/examples/arrow/compute_and_write_csv_example.cc]: @bkmgit , I'm in holiday next week, may not have time to review your code. Will do when I'm back asap.

Comment (bkmgit) [reply cpp/examples/arrow/compute_and_write_csv_example.cc]: ok

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**